app/script/auth/page/ConversationJoin.tsx

```diff
@@ -196,7 +196,24 @@ class ConversationJoin extends React.Component<CombinedProps, State> {
         .then(() => this.props.doJoinConversationByCode(this.state.conversationKey, this.state.conversationCode))
         .then(conversationEvent => this.props.setLastEventDate(new Date(conversationEvent.time)))
         .then(() => this.routeToApp())
-        .catch(error => this.props.doLogout());
+        .catch(error => {
+          if (error.label) {
+            switch (error.label) {
+              default: {
+                const isValidationError = Object.values(ValidationError.ERROR).some(errorType =>
```

Comment:
I see those 3 lines of code multiple times. How about extracting this to a util? 
Maybe a function in the `ValidationError` module?

Reply:
Jup, thought the same (actually the whole validation). I'll create a ticket for the next service run.